src/olympia/addons/models.py

```diff
@@ -237,57 +238,137 @@ def not_disabled_by_mozilla(self):
             # crucially, it prevents the
             # select_related('_current_version__autoapprovalsummary') from
             # working, because it overrides the _current_version with the one
-            # it fetches. We want translations though.
-            .only_translations()
-            # We need those joins for the queue to work without making extra
-            # queries.
-            .select_related(
-                'addonapprovalscounter',
-                'reviewerflags',
+            # it fetches. We want translations though, but only for the name.
+            .only_translations().defer(
+                *[x.name for x in Addon._meta.translated_fields if x.name != 'name']
             )
         )
-        if listed:
-            # We need those joins for the listed queues to work without making
-            # extra queries.
-            qs = qs.select_related(
-                '_current_version',
-                '_current_version__autoapprovalsummary',
-                '_current_version__reviewerflags',
-                'promotedaddon',
-            ).prefetch_related(
+        # Useful joins to avoid extra queries.
+        select_related_fields = [
+            'reviewerflags',
+            'addonapprovalscounter',
+        ]
+        if select_related_fields_for_listed:
+            # Most listed queues need these to avoid extra queries because
+            # they display the score, flags, promoted status, link to files
+            # etc.
+            select_related_fields.extend(
+                (
+                    '_current_version',
+                    '_current_version__autoapprovalsummary',
+                    '_current_version__reviewerflags',
+                    'promotedaddon',
+                )
+            )
+            qs = qs.prefetch_related(
                 '_current_version__files',
             )
-        if not show_pending_rejection:
-            if not listed:
-                # Can't combine these, because the filter is on `_current_version`
-                # which doesn't make sense for unlisted. It needs to be manually added
-                # to the same filter/Q object that joins on `versions` for unlisted.
-                raise ImproperlyConfigured(
-                    'Can not combine show_pending_rejection=False with listed=False'
-                )
+        qs = qs.select_related(*select_related_fields)
+
+        if exclude_listed_pending_rejection:
             qs = qs.filter(
                 _current_version__reviewerflags__pending_rejection__isnull=True
             )
         if not admin_reviewer:
-            if admin_content_review:
+            if content_review:
                 qs = qs.exclude(reviewerflags__needs_admin_content_review=True)
+            elif theme_review:
+                qs = qs.exclude(reviewerflags__needs_admin_theme_review=True)
             else:
                 qs = qs.exclude(reviewerflags__needs_admin_code_review=True)
```

Comment:
is it any slower to just always exclude all of the`reviewerflags__needs_admin_xxx_review.` if not `admin_reviewer`?  If so, less complexity and we don't need to pass in `content_review` and `theme_review`

Reply:
probably not that much, but it wouldn't match the behavior we have - the fields were separated on purpose, so I didn't want to touch that logic.